### PR TITLE
[App Service] `az functionapp create`: Add support for creating a webapp joined to a vnet

### PR DESCRIPTION
**Pull request by StrawnSC**
**Description**

Adds `--vnet` and `--subnet` arguments to `az webapp/functionapp create` to allow creating webapps and functionapps that are joined to a virtual network 
Also refactors `az webapp/functionapp vnet-integration add` to use the new API for integrating vnets with apps 

**Testing Guide**

`azdev test WebappNetworkConnectionTests --live`
`azdev test FunctionappNetworkConnectionTests --live`                                                         

**History Notes**

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [X] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [X] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [X] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
App Service

File: src/azure-cli/azure/cli/command_modules/appservice/_params.py [hunk 1]
```diff
@@ -153,6 +153,8 @@ def load_arguments(self, _):
                    completer=get_resource_name_completion_list('Microsoft.Web/serverFarms'),
                    help="name or resource id of the app service plan. Use 'appservice plan create' to get one",
                    local_context_attribute=LocalContextAttribute(name='plan_name', actions=[LocalContextAction.GET]))
+        c.argument('vnet', help="Name or resource ID of the virtual network. If there are multiple vnets of the same name across different resource groups, use vnet resource id to specify which vnet to use. If vnet name is used, by default, the vnet in the same resource group as the webapp will be used. Must be used with --subnet argument.")
```
Comment (panchagnula) [inline src/azure-cli/azure/cli/command_modules/appservice/_params.py:156]: can you check with the VNET intergration command. - if this text needs to say regional VNET - for clarity?

Comment (StrawnSC) [reply src/azure-cli/azure/cli/command_modules/appservice/_params.py:156]: Looks like the vent-integration commands call it a "regional vnet". I'll go ahead and change it here

**Review by panchagnula — APPROVED** [review]
LGTM

**Pull request merged**